### PR TITLE
Correctly filter revision fields for story revisions

### LINKED ISSUE #12547: Restore from autosave does not revert changes in pages (but story data is restored)
## Bug Description

Restoring an autosaved webstory that is different from the local version does not work in reverting changes to pages. However, a change in the story data (like title) is reverted !

## Steps to Reproduce
The following steps consistently reproduce the problem:

1. Create a new story with 1 page
2. Publish it with poster and description etc
3. Add another page and move it to first position
4. Update this story
5. Now add a third page, change the title & wait for a min to create the autosave session key
6. exit to dashboard without saving to DB
7. reopen the now published & updated story
8. you will get two overlays - one for restoring from local backup (dismiss this) and another for restoring from saved version ...
9. Here, view the different versions and restore the latest (3 slides)

ISSUE: the version with 2 slides is restored instead of the last saved version

## Additional Context

This issue was observed in #12521 as well.

- Plugin Version: `main branch` as of 10/25/2022
- WordPress Version: 6.0.3
- Operating System: Win10
- Browser: Chrome

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

When restoring a story from an autosave, the story content was not correctly restored.

This is because when filtering `_wp_post_revision_fields`, the post type can either be `web-story` or `revision`, and we didn't check for the latter.

## Summary

Correctly filter revision fields for story revisions by checking the post parent.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Publish a new story.
2. Add some more content to it, then wait for to be autosaved to the database
3. Refresh the page, then click on the (second) dialog CTA to view the autosaves
4. On the revisions comparison screen, restore the latest version
5. Verify that the correct content has been restored

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12547

### CODE REVIEW

File: tests/phpunit/integration/tests/Story_Revisions.php [hunk 1]
```diff
@@ -102,4 +104,63 @@ public function data_test_revisions_to_keep(): array {
...
+			[
+				'post_type'   => Story_Post_Type::POST_TYPE_SLUG,
+				'post_parent' => 0,
+			]
+		);
+
+		$this->assertArrayHasKey( 'post_content_filtered', $actual );
+	}
+
+	public function test_filter_revision_fields_story_post_type_revision(): void {
```
Comment (spacedmonkey) [inline tests/phpunit/integration/tests/Story_Revisions.php:157]: Add cover notiations.

File: includes/Story_Revisions.php [hunk 2]
```diff
@@ -101,13 +107,22 @@ public function revisions_to_keep( $num ): int {
...
 	public function filter_revision_fields( $fields, array $story ) {
 		if ( ! \is_array( $fields ) ) {
 			return $fields;
 		}
 
-		if ( $this->story_post_type->get_slug() === $story['post_type'] ) {
+		if (
+			$this->story_post_type->get_slug() === $story['post_type'] ||
+			(
+				'revision' === $story['post_type'] &&
```
Comment (spacedmonkey) [inline includes/Story_Revisions.php:121]: Should this also include autosaves?

**Review by spacedmonkey — APPROVED** [review]
Some nitpic, like phpdocs on tests but otherwise, looking good.

File: includes/Story_Revisions.php (hunk 2 above)
Comment (swissspidy) [reply includes/Story_Revisions.php:121]: I don‘t think such a case exists

**Review by timarney — APPROVED** [review]

File: tests/phpunit/integration/tests/Story_Revisions.php (hunk 1 above)
Comment (swissspidy) [reply tests/phpunit/integration/tests/Story_Revisions.php:157]: Good call! Just noticed that the `coversDefaultClass` annotation was wrong here.

**Review by spacedmonkey — APPROVED** [review]

**Comment by kkalarickal** [comment]
Confirmed the fix by restoring from a saved version and verifying that the proper version was restored.

Comparing revisions

Restored web story with 3 pages

**Comment by kkalarickal** [comment]
Further tests:

Confirmed that going back in the revision history and restoring a previous version was successful.

Also confirmed that comparing two arbitrary versions and restoring between them works as expected.

Verified that restoration works across user-profiles: meaning when one user creates the original story, and another takes over editing, the revision mismatch is caught and restoring someone else's version is possible.

Verified that with each restoration, the revision count within Document panel keeps incrementing as expected !

**Pull request merged**